From b09705b1375d1f7a3ed5ff01d2a006af2cde13c1 Mon Sep 17 00:00:00 2001
From: Zhou Peng-B04994 <eagle.zhou@nxp.com>
Date: Tue, 24 Oct 2017 15:47:19 +0800
Subject: [PATCH 2678/5242] MLK-16671-3 - [i.MX8QXP/Malone]: Add vpu malone
 decoder driver

commit  f6c88721709a448434d4213520d224007c7cc5d9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Refine makefile to fix yocto build issue:
Remove redundant space after -D and -I

Signed-off-by: Zhou Peng-B04994 <eagle.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-malone/Makefile |  284 +++++++++++++++++++--------------------
 1 file changed, 142 insertions(+), 142 deletions(-)

diff --git a/drivers/mxc/vpu-malone/Makefile b/drivers/mxc/vpu-malone/Makefile
index 86af231..9c46537 100755
--- a/drivers/mxc/vpu-malone/Makefile
+++ b/drivers/mxc/vpu-malone/Makefile
@@ -33,150 +33,150 @@ MAL_KERN_OBJECTS += Malone_Firmware/PAL/pal.o
 #---------------------------------------------------------------------------
 # Build flags
 
-DEFINES = -D DTV_GATHER_PERF_METRICS \
-		  -D MVD_DTV_USERDATA \
-		  -D MVD_WAIT_BOB_INACTIVE \
-		  -D DECLIB_FORCE_HW_STOP \
-		  -D MVD_NO_BSDMA_SAFETY_MARGIN \
-		  -D MVD_CQ_ENABLE_REFILL \
-		  -D MVD_SPP_HW_GOULOMB \
-		  -D SVC_SFA_ADD_ERROR_CHECKING \
-		  -D MVC_SFA_ADD_ERROR_CHECKING \
-		  -D SVC_SPP_SAVE_CTX_PER_VCL_NAL \
-		  -D MVD_CQ_CQSR \
-		  -D AVC_SUPPORT_THRU_MVC \
-		  -D MVC_ERROR_CONTROL_INSERT_SKIP_START_CONTROLS \
-		  -D DECLIB_CTX_FLUSH_AFTER_SAVE \
-		  -D DECLIB_SERVICE_EOS \
-		  -D MVD_PERF_MEASURE \
-		  -D VC1_ENABLED \
-		  -D HEVC_ENABLED \
-		  -D HEVC_CM_WORKAROUND \
-		  -D HEVC_NEW_OUTPUT_TRIGGER \
-		  -D HEVC_ALL_PICS_REF \
-		  -D HEVC_SCAL_LIST_USE_YCRCB_XREF \
-		  -D HEVC_SFA_ADD_ERROR_CHECKING \
-		  -D MVD_DFE_DBG \
-		  -D HEVC_JVT_MODEL=100 \
-		  -D PAL_CLOCK_API \
-		  -D SVC_ADDITIONAL_DEBUG \
-		  -D DIAG_SUPPORT_ENABLED \
-		  -D ENABLE_PERF_TIMER \
-		  -D FW_API_VERSION=19 \
-		  -D GLOBAL_USE_RUN_TIME_CFG \
-		  -D ENABLE_TRACE_IN_RELEASE=0 \
-		  -D YES=1 \
-		  -D NO=0 \
-		  -D NONE=0 \
-		  -D NUP=1 \
-		  -D UCOS=2 \
-		  -D UCOS3=3 \
-		  -D RTOS=0 \
-		  -D USE_DECODER \
-		  -D ARM=0 \
-		  -D MIPS=1 \
-		  -D X86=2 \
-		  -D OR1K=3 \
-		  -D CPU=0 \
-		  -D NO_AL=0 \
-		  -D CNXT_KAL=1 \
-		  -D NXP_OSAL=2 \
-		  -D OSAL=0 \
-		  -D ARM926=0 \
-		  -D ARMR5=1 \
-		  -D ARMA53=2 \
-		  -D ARM_CPU_TYPE=2 \
-		  -D ADS=0 \
-		  -D RVDS=1 \
-		  -D GNU_MIPS=2 \
-		  -D GNU_MIPS_LNX=3 \
-		  -D GNU_ARM=4 \
-		  -D GNU_ARM_SOURCERY=5 \
-		  -D GNU_X86=6 \
-		  -D WIN_X86=7 \
-		  -D DS5=8 \
-		  -D GNU_OR32=9 \
-		  -D GNU_ARM_LINARO=10 \
-		  -D GNU_OR1K=11 \
-		  -D TOOLSET=10 \
-		  -D NO_DEBUG=0 \
-		  -D BUILD_DEBUG=1 \
-		  -D ARRAY_DEBUG=2 \
-		  -D FULL_DEBUG=3 \
-		  -D DEBUG_CAPS=0 \
-		  -D GENTB_PLATFORM=0 \
-		  -D WIN_LIB=1 \
-		  -D GEN_TB_ENC=2 \
-		  -D TARGET_PLATFORM=0 \
-		  -D VIDEO_TRANS=0 \
-		  -D GTB_TRANS=1 \
-		  -D GTB_DEC=2 \
-		  -D WINDSOR_LIB=3 \
-		  -D GTB_ENC=4 \
-		  -D MEDIA_DEC=5 \
-		  -D MEDIA_LIB=6 \
-                  -D VPU_TEST_APP=7 \
-		  -D TARGET_APP=7 \
-		  -D PAL_CLOCK_API \
-		  -D SVC_ADDITIONAL_DEBUG \
-		  -D DIAG_SUPPORT_ENABLED \
-		  -D ENABLE_PERF_TIMER \
-		  -D FW_API_VERSION=19 \
-		  -D GLOBAL_USE_RUN_TIME_CFG \
-		  -D DISABLE_TRACE \
-		  -D ENABLE_TRACE_IN_RELEASE=0 \
-		  -D YES=1 \
-		  -D NO=0 \
-		  -D NONE=0 \
-		  -D NUP=1 \
-		  -D UCOS=2 \
-		  -D UCOS3=3 \
-		  -D RTOS=0 \
-		  -D USE_DECODER \
-		  -D CHIP=0 \
-		  -D EMULATION=1 \
-		  -D HAPS=2 \
-		  -D SIMULATION=3 \
-		  -D CMODEL=4 \
-		  -D TARGET_LEVEL=0 \
-		  -D SVC_DISABLED=0 \
-		  -D SVC_ENABLED=1 \
-		  -D SVC_SUPPORT=0 \
-		  -D MVC_DISABLED=0 \
-		  -D MVC_ENABLED=1 \
-		  -D MVC_SUPPORT=1 \
-		  -D SFA_DISABLED=0 \
-		  -D SFA_ENABLED=1 \
-		  -D SFA_SUPPORT=1 \
-		  -D CNXT_HW=0 \
-		  -D NXP_HW=1 \
-		  -D HWLIB=1 \
-		  -D DTV=0 \
-		  -D STB=1 \
-		  -D PLAYMODE=0 \
-		  -D STANDARD=0 \
-		  -D REBOOT=1 \
-		  -D BOOT_ARCH=0 \
-		  -D TBPLAYER_FLOW_CHANGE_ON_REF_FRMS \
-		  -D PULSAR_MERGE \
-		  -D FSLCACHE_ENABLED \
-		  -D DECLIB_ENABLE_DFE -D DECLIB_ENABLE_DBE \
-		  -D DECLIB_ENABLE_DCP -D MVD_DCP_DYNAMIC_CONFIG \
-		  -D DECLIB_4K_SUPPORTED -D HEVC_LEVEL_5PT0_SUPPORT \
-		  -D PLAYER_LOCAL_THREAD \
-		  -D DECLIB_ISR_IN_THREAD_CTX \
-		  -D JPG_ENABLED \
-		  -D JPGD_AUTO_DOWN_SCALE \
-		  -D SPARK_ENABLED \
-		  -D RV_ENABLED \
-		  -D VP6_ENABLED \
-		  -D VP8_ENABLED \
-		  -D JPG_DPV_ENABLED \
-		  -D MALONE_64BIT_ADDR \
-		  -D DISABLE_TRACE
+DEFINES = -DDTV_GATHER_PERF_METRICS \
+		  -DMVD_DTV_USERDATA \
+		  -DMVD_WAIT_BOB_INACTIVE \
+		  -DDECLIB_FORCE_HW_STOP \
+		  -DMVD_NO_BSDMA_SAFETY_MARGIN \
+		  -DMVD_CQ_ENABLE_REFILL \
+		  -DMVD_SPP_HW_GOULOMB \
+		  -DSVC_SFA_ADD_ERROR_CHECKING \
+		  -DMVC_SFA_ADD_ERROR_CHECKING \
+		  -DSVC_SPP_SAVE_CTX_PER_VCL_NAL \
+		  -DMVD_CQ_CQSR \
+		  -DAVC_SUPPORT_THRU_MVC \
+		  -DMVC_ERROR_CONTROL_INSERT_SKIP_START_CONTROLS \
+		  -DDECLIB_CTX_FLUSH_AFTER_SAVE \
+		  -DDECLIB_SERVICE_EOS \
+		  -DMVD_PERF_MEASURE \
+		  -DVC1_ENABLED \
+		  -DHEVC_ENABLED \
+		  -DHEVC_CM_WORKAROUND \
+		  -DHEVC_NEW_OUTPUT_TRIGGER \
+		  -DHEVC_ALL_PICS_REF \
+		  -DHEVC_SCAL_LIST_USE_YCRCB_XREF \
+		  -DHEVC_SFA_ADD_ERROR_CHECKING \
+		  -DMVD_DFE_DBG \
+		  -DHEVC_JVT_MODEL=100 \
+		  -DPAL_CLOCK_API \
+		  -DSVC_ADDITIONAL_DEBUG \
+		  -DDIAG_SUPPORT_ENABLED \
+		  -DENABLE_PERF_TIMER \
+		  -DFW_API_VERSION=19 \
+		  -DGLOBAL_USE_RUN_TIME_CFG \
+		  -DENABLE_TRACE_IN_RELEASE=0 \
+		  -DYES=1 \
+		  -DNO=0 \
+		  -DNONE=0 \
+		  -DNUP=1 \
+		  -DUCOS=2 \
+		  -DUCOS3=3 \
+		  -DRTOS=0 \
+		  -DUSE_DECODER \
+		  -DARM=0 \
+		  -DMIPS=1 \
+		  -DX86=2 \
+		  -DOR1K=3 \
+		  -DCPU=0 \
+		  -DNO_AL=0 \
+		  -DCNXT_KAL=1 \
+		  -DNXP_OSAL=2 \
+		  -DOSAL=0 \
+		  -DARM926=0 \
+		  -DARMR5=1 \
+		  -DARMA53=2 \
+		  -DARM_CPU_TYPE=2 \
+		  -DADS=0 \
+		  -DRVDS=1 \
+		  -DGNU_MIPS=2 \
+		  -DGNU_MIPS_LNX=3 \
+		  -DGNU_ARM=4 \
+		  -DGNU_ARM_SOURCERY=5 \
+		  -DGNU_X86=6 \
+		  -DWIN_X86=7 \
+		  -DDS5=8 \
+		  -DGNU_OR32=9 \
+		  -DGNU_ARM_LINARO=10 \
+		  -DGNU_OR1K=11 \
+		  -DTOOLSET=10 \
+		  -DNO_DEBUG=0 \
+		  -DBUILD_DEBUG=1 \
+		  -DARRAY_DEBUG=2 \
+		  -DFULL_DEBUG=3 \
+		  -DDEBUG_CAPS=0 \
+		  -DGENTB_PLATFORM=0 \
+		  -DWIN_LIB=1 \
+		  -DGEN_TB_ENC=2 \
+		  -DTARGET_PLATFORM=0 \
+		  -DVIDEO_TRANS=0 \
+		  -DGTB_TRANS=1 \
+		  -DGTB_DEC=2 \
+		  -DWINDSOR_LIB=3 \
+		  -DGTB_ENC=4 \
+		  -DMEDIA_DEC=5 \
+		  -DMEDIA_LIB=6 \
+                  -DVPU_TEST_APP=7 \
+		  -DTARGET_APP=7 \
+		  -DPAL_CLOCK_API \
+		  -DSVC_ADDITIONAL_DEBUG \
+		  -DDIAG_SUPPORT_ENABLED \
+		  -DENABLE_PERF_TIMER \
+		  -DFW_API_VERSION=19 \
+		  -DGLOBAL_USE_RUN_TIME_CFG \
+		  -DDISABLE_TRACE \
+		  -DENABLE_TRACE_IN_RELEASE=0 \
+		  -DYES=1 \
+		  -DNO=0 \
+		  -DNONE=0 \
+		  -DNUP=1 \
+		  -DUCOS=2 \
+		  -DUCOS3=3 \
+		  -DRTOS=0 \
+		  -DUSE_DECODER \
+		  -DCHIP=0 \
+		  -DEMULATION=1 \
+		  -DHAPS=2 \
+		  -DSIMULATION=3 \
+		  -DCMODEL=4 \
+		  -DTARGET_LEVEL=0 \
+		  -DSVC_DISABLED=0 \
+		  -DSVC_ENABLED=1 \
+		  -DSVC_SUPPORT=0 \
+		  -DMVC_DISABLED=0 \
+		  -DMVC_ENABLED=1 \
+		  -DMVC_SUPPORT=1 \
+		  -DSFA_DISABLED=0 \
+		  -DSFA_ENABLED=1 \
+		  -DSFA_SUPPORT=1 \
+		  -DCNXT_HW=0 \
+		  -DNXP_HW=1 \
+		  -DHWLIB=1 \
+		  -DDTV=0 \
+		  -DSTB=1 \
+		  -DPLAYMODE=0 \
+		  -DSTANDARD=0 \
+		  -DREBOOT=1 \
+		  -DBOOT_ARCH=0 \
+		  -DTBPLAYER_FLOW_CHANGE_ON_REF_FRMS \
+		  -DPULSAR_MERGE \
+		  -DFSLCACHE_ENABLED \
+		  -DDECLIB_ENABLE_DFE -DDECLIB_ENABLE_DBE \
+		  -DDECLIB_ENABLE_DCP -DMVD_DCP_DYNAMIC_CONFIG \
+		  -DDECLIB_4K_SUPPORTED -DHEVC_LEVEL_5PT0_SUPPORT \
+		  -DPLAYER_LOCAL_THREAD \
+		  -DDECLIB_ISR_IN_THREAD_CTX \
+		  -DJPG_ENABLED \
+		  -DJPGD_AUTO_DOWN_SCALE \
+		  -DSPARK_ENABLED \
+		  -DRV_ENABLED \
+		  -DVP6_ENABLED \
+		  -DVP8_ENABLED \
+		  -DJPG_DPV_ENABLED \
+		  -DMALONE_64BIT_ADDR \
+		  -DDISABLE_TRACE
 
 MALONE_KERN_DEFINEFLAGS = $(DEFINES)
-MALONE_KERN_DEFINEFLAGS += -D VPU_KERNEL_BUILD
+MALONE_KERN_DEFINEFLAGS += -DVPU_KERNEL_BUILD
 
 EXTRA_CFLAGS += $(MALONE_KERN_DEFINEFLAGS)
 EXTRA_CFLAGS += $(MALONE_KERN_HEADER)
-- 
1.7.9.5

